BlueprintUI/Sources/Layout/Column.swift

```diff
@@ -43,13 +43,48 @@ public struct Column: StackElement {
 
     public var children: [(element: Element, traits: StackLayout.Traits, key: AnyHashable?)] = []
 
-    public private(set) var layout = StackLayout(
-        axis: .vertical,
-        alignment: ColumnAlignment.leading.stackAlignment
-    )
+    public private(set) var layout: StackLayout = .defaultColumn
 
     public init() {}
 
+    /// Initializer using result builder to declaritively build up a stack.
+    /// - Parameters:
+    ///   - layout: A StackLayout describing the column's layout
+    ///   - elements: A block containing all elements to be included in the stack.
+    /// - Note: If element is a StackChild, then traits and key will be pulled from the element, otherwise defaults are passed through.
+    public init(
+        layout: StackLayout = .defaultColumn,
+        @ElementBuilder<StackChild> elements: () -> [StackChild]
+    ) {
+        self.init(elementsBuilder: elements)
+        self.layout = layout
+    }
+
+    /// Initializer using result builder to declaritively build up a stack.
+    /// - Parameters:
+    ///   - alignment: Determines the cross-axis layout
+    ///   - verticalUnderflow: Determines the layout when there is extra free space available.
+    ///   - verticalOverflow: Determines the layout when there is not enough space to fit all children as measured.
+    ///   - horizontalAlignment: Specifies how children will be aligned horizontally.
+    ///   - minimumVerticalSpacing: Spacing in between elements.
+    ///   - elements: A block containing all elements to be included in the stack.
+    /// - Note: If element is a StackChild, then traits and key will be pulled from the element, otherwise defaults are passed through.
+    public init(
+        alignment: StackLayout.Alignment = StackLayout.defaultColumn.alignment,
+        underflow: StackLayout.UnderflowDistribution = StackLayout.defaultColumn.underflow,
+        overflow: StackLayout.OverflowDistribution = StackLayout.defaultColumn.overflow,
+        alignment: ColumnAlignment = StackLayout.defaultColumn.columnAlignment,
+        minimumSpacing: CGFloat = StackLayout.defaultColumn.minimumSpacing,
+        @ElementBuilder<StackChild> elements: () -> [StackChild]
```

Comment:
I'm not in favor of this, but I'll mention that there's an alternative to passing all these params in the initializer: you can pass the stack into the builder closure as an `inout` param like the existing configure blocks do, to get a hybrid approach.

```swift
Row { row in
  row.alignment = .leading

  Foo()
  Bar()
}
```

It might require a builder method like this, but it seems to work!
```swift
static func buildExpression(_: Void) -> Component {
  []
}
```

Reply:
This is clever, although as we discussed on our call let's start with enumerating everything in the initializer and re-evaluate if that ends up not working out for us!

I wish this proposed solution didn't require `_ in` when you don't need it, although I suppose we could provide a second initializer that doesn't pass anything in!

Reply:
Do builders allow passing parameters? I thought they didn't; or at least last I checked this didn't work..

Reply:
(If this does work, it's kinda nice, IMO – as mentioned we can provide two inits!)

Reply:
one issue here is API confusion on what happens if you call row.add(...) inside the block? What order will the element be added?

Reply:
IMO let's leave this out of this first iteration. If we find the initializer parameters to be too cumbersome we can consider adding a second init with this syntax (or the other alternatives we discussed).

Reply:
> one issue here is API confusion on what happens if you call row.add(...) inside the block? What order will the element be added?

Very good point!